eclipse-plugin/eclipse/cc.codechecker.eclipse.plugin/src/cc/codechecker/plugin/views/report/details/BugPathListView.java

```diff
@@ -72,8 +72,11 @@ public void setName(String value) {
         setPartName(value);
     }
 
-    public void clear() {
-        viewer.setInput(null);
+    public void clean(boolean delegate) {
+        if(viewer != null && !delegate) {
+            viewer.setInput(null);
+            viewer.refresh();
+        }
```

Comment:
I understand the reason for cleaning the viewer only on certain conditions - but I still can't see why we handle this condition (which is unrelated to clearing the view) inside the clear function. You said now that the reason behind the condition is a different call-stack / thread, which isn't visible at all in this condition.

Think about writing a doc comment for this function (or write one for it). The intent and side effect of a function named clean() is easy to guess and clear. A function named clear(boolean) is already harder to understand. 
And if I read that it actually won't do anything when given a true parameter, I would start second guessing the API docs.